app/data_models/supplementary_data_store.py

```diff
@@ -115,10 +115,28 @@ def get_data(
         it would be get_data(identifier="some_list", selectors=["identifier"], list_item_id=list_item_id-1)
         """
         if self.is_data_repeating(identifier) and not list_item_id:
```

Comment:
We can revisit this, but this is technically flawed.
- If it is repeating, and we have a list item id then validator should have prevent that (it doesn't since we have added validation for then).
- Alternatively, since SDS is immutable, we could allow self reference, which would be a case of removing the `not list_item_id` check.

Right not, validator will allow something that runner will ignore. Happy to keep it as is for now and revisit if it becomes an issue.

Reply:
Ideally, I would like to finish this off before I leave, maybe just revisit and improve after?

Reply:
Ya, it's not something we will do now. Highlighted it for awareness that there is a small gap.